### ISSUE TITLE
No Python output if console.log() is called after Python code

### ISSUE THREAD

**Issue by marimeireles** [pyscript/pyscript#712]
### Checklist

- [X] I added a descriptive title
- [X] I searched for other issues and couldn't find a solution or duplication
- [X] I already searched in Google and didn't find any good information or help

### What happened?

Working on #622 I found a problem that I think it's coming from upstream `pyscript.py`.
When you add a js print after python code that should be printing in the screen, nothing is printed. To reproduce try the following snippet:

```
from datetime import datetime
now = datetime.now()
now.strftime("%m/%d/%Y, %H:%M:%S")
console.log("this is coming from js")
```

The string "this is coming from js" shows up in the console normally, but the python output in the page remains empty.
E.g using the `hello_world` example, this is all I see in my screen:

```
Hello world!
This is the current date and time, as computed by Python:
```

I'm exploring this, but feel free to add thoughts if you have any!
Thanks!

### What browsers are you seeing the problem on? (if applicable)

Firefox, Chrome

**Comment by marimeireles** [comment]
Ok, it doesn't happen if I use:

```
print('🥑')
console.log('banana 🍌')
```

**Comment by JeffersGlass** [comment]
Hi @marimeireles! It think this is actually intended (if unintuitive) behavior that I've tripped on a couple times.

When the \<py-script\> tag [evaluates](https://github.com/pyscript/pyscript/blob/fa7a97ca3087d9f5c0a1fe2787a57201a33a6dce/pyscriptjs/src/components/base.ts#L125-L134), Pyscript calls pyodide's [runPython](https://github.com/pyodide/pyodide/blob/6ca76fb725161dacf1940406f5996bc817ac00bb/src/js/api.ts#L72-L81)/[runPythonAsync](https://github.com/pyodide/pyodide/blob/6ca76fb725161dacf1940406f5996bc817ac00bb/src/js/api.ts#L173-L182), which are wrappers around Pyodide's [eval_code()](https://pyodide.org/en/stable/usage/api/python-api/code.html#pyodide.code.eval_code) and [eval_code_async()](https://pyodide.org/en/stable/usage/api/python-api/code.html#pyodide.code.eval_code_async) respectively.

By default, `eval_code()` and `eval_code_async()` return the value of the last expression, unless that expression has a trailing semicolon. (See the _return_mode_ and _quiet_trailing_semicolon_ parameters.)

This bit of code:
```python
from datetime import datetime
now = datetime.now()
now.strftime("%m/%d/%Y, %H:%M:%S")
```
Therefore returns the date formatted as a string, which then gets [written to the script's output element](https://github.com/pyscript/pyscript/blob/fa7a97ca3087d9f5c0a1fe2787a57201a33a6dce/pyscriptjs/src/components/base.ts#L137-L145).

If you `console.log(...)` to the end of that script, now the script returns the return value of console.log, which is None, so nothing gets printed.

---

I've run into a related issue with the same cause them my code ends with a function that I'm using for it's side effects that also returns a value, like `document.

```html
<!-- index.html -->
<div id="parent"></div>
```
```python
#my_script.py
from js import document
p = document.createElement('p')
p.innerText = "Hello, world!"
document.getElementById("parent").appendChild(p)
```
Shows up as:
![image](https://user-images.githubusercontent.com/1931111/186219994-a4408fe4-2031-44f8-99a9-b2b31c305910.png)

---

Regardless, I think this is not terribly clear behavior. Maybe as part of #622, the behavior of \<py-script\> tags should no longer be to write the return value to the output element? It's really more of what a repl would do, printing the value of the last expression, rather than what I think of a script doing.

**Comment by marimeireles** [comment]
Hey @JeffersGlass thanks for chiming in!
And very useful input. :)
I'm rewriting the PyScript's outputs and will take your points into consideration. Thank you!

**Comment by marimeireles** [comment]
Fixed by #749
